### PR TITLE
DSND-2871: Add structured logging charges-delta-consumer

### PR DESCRIPTION
**Pull request by crowlandsCH**
* structured logging version added
* code refactored slightly for readability
* PR dependent on structured logging PRs

[DSND-2871](https://companieshouse.atlassian.net/browse/DSND-2871)

[DSND-2871]: https://companieshouse.atlassian.net/browse/DSND-2871?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cgregoryCH — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/companieshouse/charges/delta/service/ApiClientServiceImpl.java [hunk 1]
```diff
@@ -32,93 +28,72 @@
...
             if (ex.getStatusCode() != 0) {
                 return new ApiResponse<>(ex.getStatusCode(), Collections.emptyMap());
             }
-            throw new RetryableErrorException(message, ex);
+            throw new RetryableErrorException(ERROR_RESPONSE_EXCEPTION, ex);
         } catch (Exception ex) {
-            String message = "Private API Generic exception";
-            logger.errorContext(log, message, ex, logMap);
-            throw new RetryableErrorException(message, ex);
+            LOGGER.error(PRIVATE_API_GENERIC_EXCEPTION, ex, DataMapHolder.getLogMap());
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/service/ApiClientServiceImpl.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/service/ApiClientServiceImpl.java [hunk 2]
```diff
@@ -32,93 +28,72 @@
...
-            logger.errorContext(log, "SDK exception", ex, logMap);
-            throw new RetryableErrorException("SDK Exception", ex);
+            LOGGER.error(SDK_EXCEPTION, ex, DataMapHolder.getLogMap());
+            throw new RetryableErrorException(SDK_EXCEPTION, ex);
         } catch (ApiErrorResponseException ex) {
-            String message = "Private API Error Response exception";
-            logMap.put("status", ex.getStatusCode());
-            logger.errorContext(log, message, ex, logMap);
+            DataMapHolder.get().status(String.valueOf(ex.getStatusCode()));
+            LOGGER.error(ERROR_RESPONSE_EXCEPTION, ex, DataMapHolder.getLogMap());
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/service/ApiClientServiceImpl.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 3]
```diff
@@ -141,79 +124,64 @@ private void removeBrokenFilingLinks(InternalChargeApi internalChargeApi,
...
-            throw new NonRetryableErrorException(message);
+            LOGGER.info(String.format(NON_RETRYABLE_RESPONSE_ERROR_MESSAGE,
+                    httpStatus), DataMapHolder.getLogMap());
+            throw new NonRetryableErrorException(String.format(NON_RETRYABLE_RESPONSE_ERROR_MESSAGE,
+                    httpStatus));
         } else if (!httpStatus.is2xxSuccessful()) {
-            String message = String.format("Retryable request DELETE Api Response %s", httpStatus);
-            logger.errorContext(logContext, message, null, logMap);
-            throw new RetryableErrorException(message);
+            LOGGER.error(String.format(RETRYABLE_RESPONSE_ERROR_MESSAGE, httpStatus),
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 4]
```diff
@@ -141,79 +124,64 @@ private void removeBrokenFilingLinks(InternalChargeApi internalChargeApi,
...
-            final String logContext,
-            final Map<String, Object> logMap) {
+    private void handleDeleteResponse(final HttpStatus httpStatus) {
 
         if (nonRetryableStatuses.contains(httpStatus)) {
-            String message = String.format("Non-retryable request DELETE Api Response %s",
-                    httpStatus);
-            logger.infoContext(logContext, message, logMap);
-            throw new NonRetryableErrorException(message);
+            LOGGER.info(String.format(NON_RETRYABLE_RESPONSE_ERROR_MESSAGE,
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 5]
```diff
@@ -141,79 +124,64 @@ private void removeBrokenFilingLinks(InternalChargeApi internalChargeApi,
...
-        logMap.put("chargeId", chargeId);
-
-        logger.trace(String.format("Performing a PUT with "
-                + "company number %s for contextId %s", companyNumber, logContext));
-        return apiClientService.putCharge(logContext,
-                        companyNumber,
-                        chargeId,
-                        internalChargeApi);
+                        () -> new NonRetryableErrorException("Charge Id is empty")));
+        DataMapHolder.get().companyNumber(companyNumber);
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 6]
```diff
@@ -105,14 +90,12 @@ public String processDelete(Message<ChsDelta> chsDelta) {
         final String chargeId = encoderUtil.encodeWithSha1(chargeIdOptional.orElseThrow(
                 () -> new NonRetryableErrorException("Charge Id is empty!")));
 
-        logMap.put("company_number", 0);
-        logMap.put("chargeId", chargeId);
-
-        final ApiResponse<Void> apiResponse = deleteCharge(logContext, chargeId);
+        DataMapHolder.get().companyNumber("0");
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/logging/LoggingKafkaListenerAspect.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+            LOGGER.info(LOG_MESSAGE_PROCESSED, DataMapHolder.getLogMap());
+
+            return result;
+        } catch (RetryableErrorException ex) {
+            // maxAttempts includes first attempt which is not a retry
+            if (retryCount >= maxAttempts -1){
+                LOGGER.error("Max retry attempts reached", ex, DataMapHolder.getLogMap());
+            } else {
+                LOGGER.info(String.format(EXCEPTION_MESSAGE,
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/logging/LoggingKafkaListenerAspect.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/consumer/ChargesDeltaConsumer.java [hunk 8]
```diff
@@ -48,32 +42,17 @@ public ChargesDeltaConsumer(ChargesDeltaProcessor deltaProcessor, Logger logger)
...
-                                    @Header(KafkaHeaders.OFFSET) String offset) {
-        Instant startTime = Instant.now();
+                                    @Header(KafkaHeaders.RECEIVED_PARTITION_ID) Integer partition,
+                                    @Header(KafkaHeaders.OFFSET) Long offset) {
         ChsDelta chsDelta = message.getPayload();
-        String contextId = chsDelta.getContextId();
-        logger.info(format("A new message successfully picked up from topic: %s, "
-                        + "partition: %s and offset: %s with contextId: %s",
-                topic, partition, offset, contextId));
+        DataMapHolder.get().contextId(chsDelta.getContextId());
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/consumer/ChargesDeltaConsumer.java]: addressed in most recent commit

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 9]
```diff
@@ -105,14 +90,12 @@ public String processDelete(Message<ChsDelta> chsDelta) {
...
+        DataMapHolder.get().mortgageId(chargeId);
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java]: addressed in most recent commit

**Review by cgregoryCH — APPROVED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Pull request merged**